microsoft/testsuites/core/storage.py

```diff
@@ -225,6 +230,40 @@ def verify_resource_disk_io(self, node: RemoteNode) -> None:
             "content read from file should be equal to content written to file",
         ).is_equal_to(original_text)
 
+    @TestCaseMetadata(
+        description="""
+        This test verifies the disk controller type of the VM.
+
+        Steps:
+        1. Get the disk type of the boot partition.
+        2. Compare it with disk controller type of the VM.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_platform_type=[AZURE],
+        ),
+    )
+    def verify_disk_controller_type(self, node: RemoteNode) -> None:
+        node_disk = node.features[Disk]
+
+        # Get VM's 'disk controller type' with azure api
+        vm_disk_controller_type = node_disk.get_disk_controller_type()
+
+        # Get 'disk controller type' from within VM.
+        os_disk_controller_type = node_disk.get_os_disk_controller_type()
+
+        # With certain SKUs & gen1 images 'disk_controller_type' will be 'None'
+        if not vm_disk_controller_type:
+            raise SkippedException(
+                f"VM disk_controller_type is '{vm_disk_controller_type}'."
+                f"Few SKUs& gen1 images doesn't support 'disk_controller_type'"
+            )
+
+        assert_that(
+            os_disk_controller_type,
+            "VM and OS disk_controller_types are different"
```

Comment:
It should describe as the expected results, or wrong results. The "are different" doesn't say it's expected or not. update it like, "The disk controller types of VM and OS should be the same."

Reply:
Updated the failure message to "The disk controller types of VM and OS should be the same."